node/src/rpc/rpc_reducer.rs

```diff
@@ -1,114 +1,395 @@
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher.push(RpcEffectfulAction::GlobalStateGet {
+                    rpc_id: *rpc_id,
+                    filter: filter.clone(),
+                });
+            }
+            RpcAction::StatusGet { rpc_id } => {
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher.push(RpcEffectfulAction::StatusGet { rpc_id: *rpc_id });
+            }
+            RpcAction::ActionStatsGet { rpc_id, query } => {
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher.push(RpcEffectfulAction::ActionStatsGet {
+                    rpc_id: *rpc_id,
+                    query: *query,
+                });
+            }
+            RpcAction::SyncStatsGet { rpc_id, query } => {
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher.push(RpcEffectfulAction::SyncStatsGet {
+                    rpc_id: *rpc_id,
+                    query: *query,
+                });
+            }
+            RpcAction::BlockProducerStatsGet { rpc_id } => {
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher.push(RpcEffectfulAction::BlockProducerStatsGet { rpc_id: *rpc_id });
+            }
+            RpcAction::MessageProgressGet { rpc_id } => {
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher.push(RpcEffectfulAction::MessageProgressGet { rpc_id: *rpc_id });
+            }
+            RpcAction::PeersGet { rpc_id } => {
+                let (dispatcher, state) = state_context.into_dispatcher_and_state();
+                let peers = collect_rpc_peers_info(state);
+                dispatcher.push(RpcEffectfulAction::PeersGet {
+                    rpc_id: *rpc_id,
+                    peers,
+                });
+            }
             RpcAction::P2pConnectionOutgoingInit { rpc_id, opts } => {
                 let rpc_state = RpcRequestState {
                     req: RpcRequest::P2pConnectionOutgoing(opts.clone()),
                     status: RpcRequestStatus::Init { time: meta.time() },
                     data: Default::default(),
                 };
-                self.requests.insert(*rpc_id, rpc_state);
+                state.requests.insert(*rpc_id, rpc_state);
+
+                let dispatcher = state_context.into_dispatcher();
+                dispatcher.push(P2pConnectionOutgoingAction::Init {
+                    opts: opts.clone(),
+                    rpc_id: Some(*rpc_id),
+                });
+                dispatcher.push(RpcAction::P2pConnectionOutgoingPending { rpc_id: *rpc_id });
             }
             RpcAction::P2pConnectionOutgoingPending { rpc_id } => {
-                let Some(rpc) = self.requests.get_mut(rpc_id) else {
+                let Some(rpc) = state.requests.get_mut(rpc_id) else {
```

Comment:
these should be `bug_condition!` right? The enabling condition should already be filtering out the action if the entry for that `rpc_id` cannot be found.